circup/__init__.py

```diff
@@ -409,6 +410,616 @@ def __repr__(self):
         )
 
 
+class Backend:
```

Comment:
I'd move these three classes into separate files to make this one shorter. It should be easy to do since this is a package already.

Reply:
I've started this process in the latest commit. Moving these got a little tricky because there were some variables and functions that both the backends and the click functions were using, at first I ran into problems with circular imports. I ended up moving the shared items into `shared.py` and then importing them from there in both __init__.py and backends.py. Let me know if there is a better way to solve that. 

`CPY_VERSION` in particular was toughest because it is being set somewhere as a global variable and then other code was expecting to read it's value afterward from there. That didn't seem to work with it moved into `shared.py` but I fixed one instance of it inside a backend by using get_circuitpython_version() instead of relying on the global var. I think there may be one or two more instances of that to fix when I get back to this.

The tests also will need lots of work after these changes, many of them are likely broken on the latest commit. I'll probably try to complete the rest of the requested changes and then go back and fix the tests.

Reply:
I think generally you'd have the shared stuff in `__init__` and move the update functionality to other files. `shared` is ok with me too though.